packages/babel-traverse/src/scope/index.ts

```diff
@@ -838,24 +856,39 @@ export default class Scope {
         this.isPure(node.left, constantsOnly) &&
         this.isPure(node.right, constantsOnly)
       );
-    } else if (isArrayExpression(node)) {
+    } else if (isArrayExpression(node) || isTupleExpression(node)) {
       for (const elem of node.elements) {
-        if (!this.isPure(elem, constantsOnly)) return false;
+        if (elem !== null && !this.isPure(elem, constantsOnly)) return false;
       }
       return true;
-    } else if (isObjectExpression(node)) {
+    } else if (isObjectExpression(node) || isRecordExpression(node)) {
       for (const prop of node.properties) {
         if (!this.isPure(prop, constantsOnly)) return false;
       }
       return true;
+    } else if (isDecorator(node)) {
+      return this.isPure(node.expression, constantsOnly);
     } else if (isMethod(node)) {
       if (node.computed && !this.isPure(node.key, constantsOnly)) return false;
       if (node.kind === "get" || node.kind === "set") return false;
```

Comment:
I don't get why evaluating getter/setter definition is `impure`.

Reply:
Uhm it _is_ pure. Probably whoever wrote this code was confused by the fact that accessing a getter/setter property is not pure.

Reply:
I agree. I have removed this and changed test cases accordingly.